### ISSUE TITLE
Consider server-based logging for catalogue errors

### ISSUE THREAD

**Issue by stevage** [TerriaJS/terriajs#1326]
When something goes wrong in a catalog item, we currently either console.log it or show a message, or both. I wonder if we should consider using a server-based log, so we can actually find the errors that users are encountering. 

It would be pretty easy to add a service to TerriaJS-Server (`/report-error?msg=...`) and call that. Probably some privacy issues to manage, but worth doing? Or perhaps there is a better approach, perhaps through a third party?

It just seems a little bit sub-optimal that many users could be encountering the same broken layer on NationalMap and no one would ever know about it unless someone actually emails data@pmc.

**Comment by kring** [comment]
What about logging errors to Google Analytics?  We might even do this already.

**Comment by stevage** [comment]
Huh, interesting idea. Could even just fetch a page that does nothing other than load the GA.

**Comment by kring** [comment]
> Could even just fetch a page that does nothing other than load the GA.

Not sure what you mean...?

**Comment by stevage** [comment]
Ok, well I'm not sure what you mean about "logging errors to GA". I thought one way you could do this would be:
1. Have an 'error.html' page that you call with an argument: `nationalmap.gov.au/error.html?msg=Could%20not%20load...
2. That page contains nothing more than `<html><script>...GA stuff...</script></html>`

So would log a GA request, presumably with the full URL. But maybe there's a simpler way to achieve the same thing.

**Comment by kring** [comment]
Yeah, look at the calls to `terria.analytics.logEvent` scattered throughout our code.  GA has a JavaScript API, you just call a function.
